go/simple/v2/examples/simple_example/simple_example.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"ampli-example/ampli"
+	"github.com/joho/godotenv"
+	"log"
+	"os"
+)
+
+func main() {
+	err := godotenv.Load()
+	if err != nil {
+		log.Fatalln("Error loading .env file")
+	}
+	apiKey := os.Getenv("AMPLITUDE_API_KEY")
+
+	// Initialize the Ampli instance with LoadOptions and LoadClientOptions
+	ampli.Instance.Load(ampli.LoadOptions{
+		Client: ampli.LoadClientOptions{
+			Configuration: ampli.NewConfig(apiKey),
+		},
+	})
+	defer ampli.Instance.Shutdown()
+
+	// Identify using IdentifyProperties in tracking plan
+	identifyObj := ampli.NewIdentify(16.6).SetOptionalArray([]string{"abc", "test"})
+	ampli.Instance.Identify("user_id", identifyObj)
+
+	// Group Identify using GroupProperties in tracking plan
+	groupIdentifyObj := ampli.NewGroup(true).SetOptionalString("optional-string")
+	ampli.Instance.GroupIdentify("Org", "Engineer", groupIdentifyObj)
+
+	// Set groups for user
+	ampli.Instance.SetGroup("user-id", "Org", []string{"Engineer", "DevOp"})
+
+	// Create a strongly typed event
+	eventMaxIntForTest := ampli.NewEventMaxIntForTest(6)
+	// Track the event using strongly typed method and parameters
+	ampli.Instance.EventMaxIntForTest("user-id", eventMaxIntForTest)
+	// Track the event with generic track method
+	ampli.Instance.Track("user-id", eventMaxIntForTest)
+
+	eventNoProperties := ampli.NewEventNoProperties()
+	ampli.Instance.EventNoProperties("user-id", eventNoProperties)
+	ampli.Instance.Track("user-id", eventNoProperties)
+
+	requiredObjectArray := map[string]int{"k": 15}
+	eventObjectTypes := ampli.NewEventObjectTypes(map[string]string{"key": "value"}, []interface{}{requiredObjectArray})
+	ampli.Instance.EventObjectTypes("user-id", eventObjectTypes)
+	ampli.Instance.Track("user-id", eventObjectTypes)
+
+	eventWithAllProperties := ampli.NewEventWithAllProperties([]string{"abc", "test"}, true, ampli.EventWithAllPropertiesRequiredEnumEnum1, 3, 16.4, "str").SetOptionalString("optional-string")
```

Comment:
I think to true enforce schema we need to use Java style builders. The current implementation does not enforce a schema changes statically 

For example the following schema change would continue to compile but would produce bad results.
MyEvent (v1) { prop1: string, prop2: string } 
MyEvent (v2) { prop2: string, zProp: string, } 

`ampli.NewMyEvent("prop1", "prop2")`
vs
`ampli.NewMyEvent.prop1("prop1").prop2("prop2")`

https://github.com/amplitude/ampli-examples/blob/main/jre/java/AmpliApp/src/main/java/org/example/App.java#L108-L115

Reply:
@justin-fiedler 
1. Added `builders` for 3 events - `EventNoProperties`, `EventWithAllProperties` and `EventWithOptionalProperties`
2. Event enums is now use style like `ampli.EventWithAllProperties.RequiredEnum.Enum1` instead of `ampli.EventWithAllPropertiesRequiredEnumEnum1`
3. To not pollute `ampli` namespace with intermediate builder types anonymous types are used in codegen:
```
	Builder func() interface {
		SetRequiredArray(requiredArray []string) interface {
			SetRequiredBool(requiredBool bool) interface {
				SetRequiredEnum(requiredEnum EventWithAllPropertiesRequiredEnum) interface {
					SetRequiredInteger(requiredInteger int) interface {
						SetRequiredNumber(requiredNumber float64) interface {
							SetRequiredString(requiredString string) EventWithAllPropertiesBuilder
						}
					}
				}
			}
		}
	}
```